intersectionobserver/demo.js

```diff
@@ -13,16 +13,24 @@ function loadItems(n) {
 }
 
 var intersectionObserver = new IntersectionObserver(entries => {
-  // If intersectionRatio is 0, the sentinel is out of view
-  // and we do not need to do anything.
-  if (entries[0].intersectionRatio <= 0) {
-    return;
-  }
-  loadItems(10);
-  // appendChild will move the existing element, so there is no need to
-  // remove it first.
-  scroller.appendChild(sentinel);
-  loadItems(5);
-  ChromeSamples.setStatus('Loaded up to item ' + counter);
+  // If the browser is busy while scrolling happens, multiple entries can
+  // accumulate between invocations of this callback, so we loop over
+  // them.
```

Comment:
Oh really? If the browser is busy, can layout really happen? I think layout is required to get the sentinel back below the threshold so that it can then cross the threshold again, doesn’t it?

Reply:
The task responsible for delivering notifications can be de-prioritized, which could cause notifications to accumulate.

Reply:
You can have a sequence like:

load handler
animation frame
load handler
animation frame
load handler
animation frame
IntersectionObserver notification delivery

Reply:
Of course, thank you.